Microsoft.Azure.CosmosRepository/src/ChangeFeed/DefaultConatinerChangeFeedProcessor.cs

```diff
@@ -0,0 +1,148 @@
+            IServiceProvider serviceProvider)
+        {
+            itemTypes.AreAllItems();
+            _containerService = containerService;
+            ItemTypes = itemTypes;
+            _leaseContainerProvider = leaseContainerProvider;
+            _changeFeedOptions = changeFeedOptions;
+            _logger = logger;
+            _serviceProvider = serviceProvider;
+        }
+
+        public IReadOnlyList<Type> ItemTypes { get; }
+
+        public async Task StartAsync()
+        {
+            Container itemContainer = await _containerService.GetContainerAsync(ItemTypes);
+            Container leaseContainer = await _leaseContainerProvider.GetLeaseContainerAsync();
+
+            _processor = itemContainer.GetChangeFeedProcessorBuilder<JObject>("cosmos-repository-pattern-processor",
+                    (changes, token) => OnChanges(changes, token, itemContainer.Id))
+                .WithLeaseContainer(leaseContainer)
+                .WithInstanceName(_changeFeedOptions.InstanceName)
+                .WithErrorNotification((token, exception) => OnError(exception, itemContainer.Id))
+                .Build();
+
+            _logger.LogInformation("Starting change feed processor for container {ContainerName}", itemContainer.Id);
+
+            await _processor.StartAsync();
+
+            _logger.LogInformation("Successfully started change feed processor for container {ContainerName}",
+                itemContainer.Id);
+        }
+
+        internal async Task OnChanges(IReadOnlyCollection<JObject> changes, CancellationToken cancellationToken,
+            string containerName)
+        {
+            _logger.LogDebug("Detected changes for container {ContainerName} total ({ChangesCount})",
+                containerName, changes.Count);
+
+            foreach (JObject change in changes)
+            {
+                if (!change.TryGetValue("type", out JToken type))
+                {
+                    //TODO: _logger.LogWarning(...) here? Should this be turn on/off thing? A user might not have created documents with the type field ?!
+                    continue;
+                }
+
+                Type itemType = ItemTypes.FirstOrDefault(x => x.Name == type.Value<string>());
+
+                if (itemType is null)
+                {
+                    _logger.LogDebug(
+                        "No change feed processor registered for type {ItemType} in container {ContainerName}",
+                        type.Value<string>(), containerName);
+
+                    continue;
+                }
+
+                await InvokeHandler(itemType, change, cancellationToken);
+            }
+        }
+
+        private async Task InvokeHandler(Type itemType, JObject instance, CancellationToken cancellationToken)
+        {
+            object item = instance.ToObject(itemType);
+
+            Type handlerType = null;
+
+            if (Handlers.ContainsKey(itemType) is false)
```

Comment:
Where does `Handlers` come from?

Reply:
It's just a static collection of constructed handler types, it means you only construct the generic type once.